src/System.Private.Xml/src/System/Xml/Schema/XsdDateTime.cs

```diff
@@ -538,14 +561,61 @@ public override string ToString()
         private void PrintDate(StringBuilder sb)
         {
             char[] text = new char[s_lzyyyy_MM_dd];
-            IntToCharArray(text, 0, Year, 4);
+            int year, month, day;
+            GetYearMonthDay(out year, out month, out day);
+            IntToCharArray(text, 0, year, 4);
             text[s_lzyyyy] = '-';
-            ShortToCharArray(text, s_lzyyyy_, Month);
+            ShortToCharArray(text, s_lzyyyy_, month);
             text[s_lzyyyy_MM] = '-';
-            ShortToCharArray(text, s_lzyyyy_MM_, Day);
+            ShortToCharArray(text, s_lzyyyy_MM_, day);
             sb.Append(text);
         }
 
+        private void GetYearMonthDay(out int year, out int month, out int day)
```

Comment:
Can you please add comments explaining where the implementation was from and why we wanted to have the implementation here?